load-test/scenarios/teacher-edit-pupil/teacher-edit-pupil.jmx

```diff
@@ -0,0 +1,398 @@
+              </elementProp>
+            </collectionProp>
+          </elementProp>
+          <stringProp name="HTTPSampler.domain"></stringProp>
+          <stringProp name="HTTPSampler.port"></stringProp>
+          <stringProp name="HTTPSampler.protocol"></stringProp>
+          <stringProp name="HTTPSampler.contentEncoding"></stringProp>
+          <stringProp name="HTTPSampler.path">pupil-register/pupil/edit</stringProp>
+          <stringProp name="HTTPSampler.method">POST</stringProp>
+          <boolProp name="HTTPSampler.follow_redirects">true</boolProp>
+          <boolProp name="HTTPSampler.auto_redirects">false</boolProp>
+          <boolProp name="HTTPSampler.use_keepalive">true</boolProp>
+          <boolProp name="HTTPSampler.DO_MULTIPART_POST">false</boolProp>
+          <stringProp name="HTTPSampler.embedded_url_re"></stringProp>
+          <stringProp name="HTTPSampler.connect_timeout"></stringProp>
+          <stringProp name="HTTPSampler.response_timeout"></stringProp>
+        </HTTPSamplerProxy>
+        <hashTree>
+          <ConstantTimer guiclass="ConstantTimerGui" testclass="ConstantTimer" testname="Wait-To-Add-Pupil" enabled="true">
+            <stringProp name="ConstantTimer.delay">1000</stringProp>
+          </ConstantTimer>
+          <hashTree/>
+          <BeanShellPreProcessor guiclass="TestBeanGUI" testclass="BeanShellPreProcessor" testname="BeanShell PreProcessor" enabled="true">
+            <stringProp name="filename"></stringProp>
+            <stringProp name="parameters"></stringProp>
+            <boolProp name="resetInterpreter">false</boolProp>
+            <stringProp name="script">
+String laCode = &quot;001&quot;;
+String estabNum = &quot;${__Random(1000,9999)}&quot;;
+String yearOfAlloc = &quot;${__Random(10,99)}&quot;;
+String serialNum = &quot;${__Random(100,999)}&quot;;
+
+String upnDigits = laCode + estabNum + yearOfAlloc + serialNum;
+int index= 2;
+int sum =0;
+ArrayList upnArray = new ArrayList();
+for(int i=0; i&lt;upnDigits.length(); i++){	
+	int j = ((upnDigits.charAt(i)) - 48 ) * index;
+	upnArray.add(j);
+	index++;
+}
+for(int number: upnArray){
+	sum += number;
+}
+checkLetterIndex = sum % 23;
+Map hmap = new HashMap(); 
+hmap.put(0, &quot;A&quot;);
+hmap.put(1, &quot;B&quot;);
+hmap.put(2, &quot;C&quot;);
+hmap.put(3, &quot;D&quot;);
+hmap.put(4, &quot;E&quot;);
+hmap.put(5, &quot;F&quot;);
+hmap.put(6, &quot;G&quot;);
+hmap.put(7, &quot;H&quot;);
+hmap.put(8, &quot;J&quot;);
+hmap.put(9, &quot;K&quot;);
+hmap.put(10, &quot;L&quot;);
+hmap.put(11, &quot;M&quot;);
+hmap.put(12, &quot;N&quot;);
+hmap.put(13, &quot;P&quot;);
+hmap.put(14, &quot;Q&quot;);
+hmap.put(15, &quot;R&quot;);
+hmap.put(16, &quot;T&quot;);
+hmap.put(17, &quot;U&quot;);
+hmap.put(18, &quot;V&quot;);
+hmap.put(19, &quot;W&quot;);
+hmap.put(20, &quot;X&quot;);
+hmap.put(21, &quot;Y&quot;);
+hmap.put(22, &quot;Z&quot;);
+
+String upnNuunber= hmap.get(checkLetterIndex) + upnDigits;
```

Comment:
should this be `upnNumber` ?

Reply:
also, are we sure this script always generates a valid UPN?

Reply:
Yes, this is to generate UPN and yes I am sure it work always as I tested it several times with many users.